archive/r/rust/fibonacci.rs

```diff
@@ -0,0 +1,37 @@
+const LIMIT: i64 = 92;
+
+fn fibonacci(terms: i64) {
+    if terms > LIMIT {
+        println!("The number of terms you want to calculate is too big!");
+        println!("The limit is 92.");
+    } else {
+        println!("\n");
+        let mut i = 0i64;
+        let mut a = 0i64;
+        let mut b = 1i64;
+        let mut c = 0i64;
+        while i < terms {
```

Comment:
I'm no Rust expert, but why not use a for loop here?

Reply:
Mainly a style decision.

Reply:
The `for` vs `while` debate is one I was just introduced to recently. It seems some folks think `for` has a pretty nasty syntax. In fact, I heard someone complain that C brought it into the world haha

Reply:
I come from a C++ background and I use mainly `while` loops. In Rust (and other languages like Python) the `for` loop is used more especially for iterating over a list of items.

Reply:
Is that because it’s a for each loop like python?

Reply:
Yes. Its syntax is similar.